### ISSUE TITLE
Unexpected behavior with dicts from sprig functions

### ISSUE THREAD

**Issue by DiegoLelis** [kubernetes/kops#10533]
**1. What `kops` version are you running? The command `kops version`, will display
 this information.**

Tested on 1.18.0 and 1.18.2

**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**

v1.17.14

**3. What cloud provider are you using?**
AWS

**4. What commands did you run?  What is the simplest way to reproduce this issue?**
kops toolbox template --values "kops-temp.yaml" --template "template.yaml" -v 10 --format-yaml > "kops-temp-cluster.yaml"

This doesn't work (problem described)
```
    {{ range .cluster.zones_temp }}
    {{ $aZ := dict "a" 1 "b" 2 "c" 3 "d" 4 }}
    {{ get $aZ "a" }}
    {{end}}
```

This works fine: 
```
    {{ range .cluster.zones_temp }}
    {{ $aZ := dict "a" 1 "b" 2 "c" 3 "d" 4 }}
    {{end}}
```

**5. What happened after the commands executed?**
unable to render template: template.yaml, error: template: mainTemplate:237: function "get" not defined

**6. What did you expect to happen?**
According to this all the spring functions are available: [Link](https://kops.sigs.k8s.io/operations/cluster_template/#sprig-functions)
According to [this](http://masterminds.github.io/sprig/dicts.html), this is how it was supposed to work.

**7. Please provide your cluster manifest. Execute
  `kops get --name my.example.com -o yaml` to display your cluster manifest.
  You may want to remove your cluster name and other sensitive information.**
Already provided the problematic snippet of code.

**8. Please run the commands with most verbose logging by adding the `-v 10` flag.
  Paste the logs into this report, or in a gist and provide the gist link here.**

Already did.

**9. Anything else do we need to know?**

**Comment by DiegoLelis** [comment]
It seems version 1.18 doesn't have the get function yet: [Release docs](https://github.com/kubernetes/kops/blob/release-1.18/vendor/github.com/Masterminds/sprig/functions.go#L89l)

Does anybody know a way to get the value out of a dictionary on 1.18?

**Comment by olemarkus** [comment]
This will be fixed in 1.20 through #10191

**Comment by DiegoLelis** [comment]
Found a workaround using the pluck function.
This will work: 
```
    {{ $zoneMap := dict "a" (list 0 32) "b" (list 64 96) "c" (list 128 160) "d" (list 192 224) }}
    {{ range .cluster.zones_temp }}
    {{ $currentZone := . }}
    {{ $zoneIps := pluck $currentZone $zoneMap }}
    {{ $zoneIps := index $zoneIps 0 }}
    - {{ $zoneIps }}
    {{end}}
```
